File: app/src/main/java/com/popalay/cardme/presentation/widget/LettersImageView.java
package com.popalay.cardme.presentation.widget;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v7.widget.AppCompatImageView;
import android.text.TextUtils;
import android.util.AttributeSet;

import com.amulyakhare.textdrawable.TextDrawable;
import com.amulyakhare.textdrawable.util.ColorGenerator;
import com.popalay.cardme.utils.StringUtils;

public class LettersImageView extends AppCompatImageView {

    private String currentText;

    public LettersImageView(Context context) {
        super(context);
    }

    public LettersImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public LettersImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setText(String text) {
        if(TextUtils.equals(text, currentText)) return;
        currentText = text;
        applyLetters(text);
    }

    private void applyLetters(String text) {
        final ColorGenerator generator = ColorGenerator.MATERIAL;
        final TextDrawable drawable = TextDrawable.builder()
                .buildRound(StringUtils.getFirstLetters(text), generator.getColor(text));
        setImageDrawable(drawable);
    }
}

File: app/src/main/java/com/popalay/cardme/presentation/widget/LettersImageView.kt
package com.popalay.cardme.presentation.widget

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.text.TextUtils
import android.util.AttributeSet

import com.amulyakhare.textdrawable.TextDrawable
import com.amulyakhare.textdrawable.util.ColorGenerator
import com.popalay.cardme.utils.firstLetters

class LettersImageView : AppCompatImageView {

    private var currentText: String = ""
    private val generator: ColorGenerator by lazy { ColorGenerator.MATERIAL }

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    fun setText(text: String) {
        if (TextUtils.equals(text, currentText)) return
        currentText = text
        applyLetters(text)
    }

    private fun applyLetters(text: String) {
        setImageDrawable(TextDrawable.builder().buildRound(text.firstLetters(), generator.getColor(text)))
    }
}
